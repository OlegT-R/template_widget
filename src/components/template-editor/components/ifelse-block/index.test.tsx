import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { IfElseBLock } from '.';

const node = {
  id: 'id',
  if: [
    {
      id: 'testifElseIF',
      value: 'testifElseIF',
    },
  ],
  then: [
    {
      id: 'testifElseTHEN',
      value: 'testifElseTHEN',
    },
  ],
  else: [
    {
      id: 'testifElseELSE',
      value: 'testifElseELSE',
    },
  ],
};

const onDelete = () => null;
const onSetActiveTextArea = () => null;
test('<IfElseBLock />', () => {
  const component: any = shallow(
    <IfElseBLock
      onDelete={onDelete}
      node={node}
      parentNode={[node]}
      onSetActiveTextArea={onSetActiveTextArea}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});
