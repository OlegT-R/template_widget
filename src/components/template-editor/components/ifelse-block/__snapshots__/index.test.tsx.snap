// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IfElseBLock /> 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.span>
        If
      </styled.span>
      <Styled(Button)
        onClick={[Function]}
        size="small"
        type="danger"
      >
        delete
        <Icon
          type="delete"
        />
      </Styled(Button)>
    </styled.div>
    <styled.div>
      <TextArea
        key="testifElseIF"
        node={
          Object {
            "id": "testifElseIF",
            "value": "testifElseIF",
          }
        }
        onSetActiveTextArea={[Function]}
        parentNode={
          Array [
            Object {
              "id": "testifElseIF",
              "value": "testifElseIF",
            },
          ]
        }
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.span>
        Then
      </styled.span>
    </styled.div>
    <styled.div>
      <TextArea
        key="testifElseTHEN"
        node={
          Object {
            "id": "testifElseTHEN",
            "value": "testifElseTHEN",
          }
        }
        onSetActiveTextArea={[Function]}
        parentNode={
          Array [
            Object {
              "id": "testifElseTHEN",
              "value": "testifElseTHEN",
            },
          ]
        }
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.span>
        Else
      </styled.span>
    </styled.div>
    <styled.div>
      <TextArea
        key="testifElseELSE"
        node={
          Object {
            "id": "testifElseELSE",
            "value": "testifElseELSE",
          }
        }
        onSetActiveTextArea={[Function]}
        parentNode={
          Array [
            Object {
              "id": "testifElseELSE",
              "value": "testifElseELSE",
            },
          ]
        }
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
